src/Extension/DebugExtension.php

```diff
@@ -61,4 +63,27 @@ function twig_var_dump(Environment $env, $context, ...$vars)
 
     return ob_get_clean();
 }
+
+function twig_var_dump_to_json(Environment $env, $context, ...$vars)
+{
+    if (!$env->isDebug()) {
+        return;
+    }
+
+    ob_start();
+
+    if (!$vars) {
+        $vars = [];
+        foreach ($context as $key => $value) {
+            if (!$value instanceof Template && !$value instanceof TemplateWrapper) {
+                $vars[$key] = $value;
+            }
+        }
+        echo json_encode($vars);
+    } else {
+        echo json_encode(...$vars);
```

Comment:
echo seemed like it made more sense here, but if there's a reason return is preferred I'd love to see it :) Couldn't find much on Google about it.

Reply:
Your function still returns the data, as you echo it in an output buffer that you then read to return its content. But your implementation is adding useless overhead (and misses the proper error handling to avoid leaking an open output buffer in case an exception happens)

Reply:
Excellent, thank you. That makes a lot of sense :)